src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -46,7 +45,7 @@ public JsonAdaptedPerson(Person source) {
         name = source.getName().fullName;
         phone = source.getPhone().value;
         email = source.getEmail().value;
-        isDone = source.getIsDone().value.toUpperCase(Locale.ROOT);
+        isDone = requireNonNull(source.getIsDone().value) ? "TRUE" : "FALSE";
```

Comment:
isDone.value will never be null tho, since it is of type 'boolean'. So it makes this requireNonNull thing pointless. Theres 2 options about this. One is to change the type of value in IsDone to 'Boolean' instead. Or two, do the non-null check in toModelType. Personally i prefer the latter. We established that isDone is always assumed to be false when it is not explicitly stated. So isDone.value() should never be null to begin with.

Reply:
lol this is the exact same thing ive been trying to tell you.

Reply:
Wait in the previous comment u said u personally want the isdone value to be boolean (which i already did). Now u personally want the model type to have a non-null check. Could you clarify this?

![Clarification](https://user-images.githubusercontent.com/77221700/135739199-9775626e-a0fb-4636-81ab-9a768efe508f.PNG)

Reply:
The non null check will be done on the isDone field of JsonAdaptedPerson. Which can still be null. This is bcos the this.isDone of JsonAdaptedPerson can set by 2 methods. The method u are referring to right now is the one that converts a person into a JsonAdaptedPerson. In this case, you are right, this.isDone will never be null since person.getIsDone().value() is a boolean. However, i think u have been over looking the other method which is the one with the @JsonCreator tag. This is the method that converts the storage data into a JsonAdaptedPerson. So lets say for example, the email field does not exist in the json storage. This method will basically set this.email as null. Hence why there is a need for the null check in the toModelType. Likewise, if we keep the original line 38 `this.isDone = isDone', if the isDone field does not exist in the json storage, it will be set to null as well. Hence this should get rid of the `condition isDone == null is always false` message.

Reply:
I think the reply directly above resolves the other issue in another comment thread

This comment thread was regarding the comment here. 
> One is to change the type of value in IsDone to 'Boolean' instead. Or two, do the non-null check in toModelType.

I was asking what value of isDone do you mean. When u say the type of value in IsDone do u mean isDone.value ? or isDone in the JSON ? i just wanted to need some clarification as it causes a lot of confusion.